config/static_provider_test.go

```diff
@@ -208,3 +208,13 @@ func TestPopulateForNestedMaps(t *testing.T) {
 	assert.Contains(t, err.Error(), `empty map key is ambigious`)
 	assert.Contains(t, err.Error(), `a.`)
 }
+
+func TestPopulateNonPointerType(t *testing.T) {
+	t.Parallel()
+
+	p := NewStaticProvider(42)
+	x := 13
+	err := p.Get(Root).Populate(x)
```

Comment:
Only if a value returned by a provider implements the Interface:
https://github.com/uber-go/fx/blob/dev/config/decoder.go#L371